crypto/evp/evp_lib.c

```diff
@@ -387,7 +387,7 @@ int evp_cipher_cache_constants(EVP_CIPHER *cipher)
 
 int EVP_CIPHER_get_block_size(const EVP_CIPHER *cipher)
 {
-    return cipher->block_size;
+    return cipher != NULL ? cipher->block_size : 0;
```

Comment:
I think it is a wrong place to fix this. I would check that we are not calling EVP_CIPHER_get_block_size() with NULL cipher in the cmac code.

Reply:
My feeling was that changing `EVP_CIPHER_get_block_size()` is simpler, and it makes that API more robust.

If we want to limit this to cmac, then I can add a null check below:

```C
// cmac_prov.c, line 100
static size_t cmac_size(void *vmacctx)
{
    struct cmac_data_st *macctx = vmacctx;

    return EVP_CIPHER_CTX_get_block_size(CMAC_CTX_get0_cipher_ctx(macctx->ctx));
}
```

Reply:
Please do that.

Reply:
I tend to agree with @t8m.